extensions/amp-form/0.1/amp-form.js

```diff
@@ -556,14 +556,14 @@ export class AmpForm {
       // Validity checking should always occur, novalidate only circumvent
       // reporting and blocking submission on non-valid forms.
       const isValid = checkUserValidityOnSubmission(this.form_);
-      if (this.shouldValidate_ && !isValid) {
+      if (this.shouldValidate_) {
```

Comment:
Yeah the standard validator function has that behavior, but the `vsync` in this block doesn't call the standard function, it calls the amp-form's validator class, which may or may not be the standard validator. I checked and the custom `[ShowFirstOnSubmitValidator](https://github.com/cvializ/amphtml/blame/master/extensions/amp-form/0.1/form-validators.js#L282)` calls `this.hideAllValidations();` without checking if the  invalid first or not. So calling the `checkValidity` function as implemented in this PR could have visual side effects, when you expected to simply get a boolean value without causing side effects. Or am I misunderstanding something?